src/VisualQueryBuilder/components/LabelParamEditor.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState } from 'react';
+
+import { DataSourceApi, SelectableValue, toOption } from '@grafana/data';
+import { Select } from '@grafana/ui';
+
+import { getOperationParamId } from './OperationEditor';
+import { QueryBuilderLabelFilter, QueryBuilderOperationParamEditorProps, VisualQuery, VisualQueryModeller } from '../types';
+
+
+export const LabelParamEditor = ({
+  onChange,
+  index,
+  operationId,
+  value,
+  query,
+  datasource,
+  queryModeller,
+}: QueryBuilderOperationParamEditorProps) => {
+  const [state, setState] = useState<{
+    options?: SelectableValue[];
+    isLoading?: boolean;
+  }>({});
+
+  return (
+    <Select
+      inputId={getOperationParamId(operationId, index)}
+      autoFocus={value === '' ? true : undefined}
+      openMenuOnFocus
+      onOpenMenu={async () => {
+        setState({ isLoading: true });
+        const options = await loadGroupByLabels(query, datasource, queryModeller);
+        setState({ options, isLoading: undefined });
+      }}
+      isLoading={state.isLoading}
+      allowCustomValue
+      noOptionsMessage="No labels found"
+      loadingMessage="Loading labels"
+      options={state.options}
+      value={toOption(value as string)}
```

Comment:
Value can be undefined, so this is another type to fix and a possible source for bugs.

Reply:
Added to https://github.com/grafana/grafana-experimental/issues/101